v2/contribute/decisions/thirdparty/0007-check-for-enabled-recipes.mdx

```diff
@@ -0,0 +1,46 @@
+---
+id: "0007"
+title: Should recipe check for enabled recipes in the core
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Should recipe check for enabled recipes in the core
+
+<DecisionHeader status="proposed" lastUpdate="2022-12-08" created="2022-12-08" deciders={["rishabhpoddar", "sattvikc"]} proposedBy={["sattvikc"]} />
+
+## Context and Problem Statement
+
+Multitenancy recipe now supports enabling and disabling of recipes per tenant. But the recipes are also initialized statically in the Supertokens init.
+
+Should each of the recipe check if the recipe is enabled in the core?
+
+## Considered Options
+
+* Recipes should not check
+* Recipes should check
+
+
+## Decision Outcome
+
+Chosen option: **Recipes should check**, because
+
+* This will ensure consistency in behaviour as per user configuration and SDK behaviour
+
+## Pros and Cons of the Options
+
+### Recipes should not check
+
+Recipes functions/APIs don't check for this flag, and, all the statically configured recipes will function as defined, even if the user has disabled them in the core.
+
+<ArgumentCon> Inconsistent SDK behaviour with the user configuration. For example, a user might be able to create an emailpassword account and use the application, which a particular tenant intends to disable, just because it was statically declared for use with a different tenant. </ArgumentCon>
+
+### Recipes should check
+
+The recipe functions/APIs check if the recipe is enabled in the core and then continue it's execution
```

Comment:
I think only the APIs should check for this and not the functions - we should still allow the dev to call the functions manually for a tenant even if they have not enabled that recipe for that tenant. The reason for this is that it provides more flexibility and also has no security issues. The reason for checking in API is cause those are public facing and there pose a security risk.